File: 0238-product-of-array-except-self/0238-product-of-array-except-self.js
var productExceptSelf = function (nums) {
    let arr = new Array(nums.length)

    let left = 1
    for(let i = 0; i < nums.length; i++){
        arr[i] = left;
        left *= nums[i]
    }

    let right = 1
    for(let j = nums.length-1; j >= 0; j--){
        arr[j] *= right
        right *= nums[j]
    }

    return arr
};

File: 0238-product-of-array-except-self/0238-product-of-array-except-self.cs
public class Solution {
    public int[] ProductExceptSelf(int[] nums) {
        int[] arr = new int[nums.Length];

        int left = 1;
        for(int i = 0; i < nums.Length; i++)
        {
            arr[i] = left;
            left *= nums[i];
        }

        int right = 1;
        for(int j = nums.Length-1; j >= 0; j--)
        {
            arr[j] *= right;
            right *= nums[j];
        }

        return arr;
    }
}